src/main.rs

```diff
@@ -205,6 +204,21 @@ fn parse_opts_into_struct(
         }
     }
 
+    // Rebuild the filter set to apply any custom brute settings
+    let mut new_filters = IndexSet::new();
+    for mut f in opts.filter.drain(..) {
+        if let FilterStrategy::Brute { num_lines, level } = &mut f {
+            if let Some(new_lines) = matches.get_one::<usize>("brute-lines") {
+                *num_lines = *new_lines;
+            }
+            if let Some(new_level) = matches.get_one::<i64>("brute-level") {
+                *level = *new_level as u8;
+            }
+        }
+        new_filters.insert(f);
+    }
+    opts.filter = new_filters;
```

Comment:
This feels a bit awkward but I'm not sure if there's a better way to do it. Values in an index set aren't mutable like they are in a vec.